### PR TITLE
restructure X to modularize cosmos sdk X modules

### PR DESCRIPTION
**Pull request by JaapWijnen**

### CODE REVIEW

**Review by ratranqu — COMMENTED** [review]
Overall I think it's fine.

Should we move to `X/Auth` instead of `X/XAuth` now for module names rather than later?

**Comment by JaapWijnen** [comment]
@ratranqu renamed the modules to without the X prefix!

**Review by ratranqu — APPROVED** [review]
looks good to me.

**Comment by ratranqu** [comment]
@paulofaria : if you hv to review today, would be awesome, else no worries. The intent is to start to decouple the modules, with minimal impact to the implicit map we have between go and swift code.

File: Sources/Cosmos/Types/Simulation/Account.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+//  File.swift
```
Comment (paulofaria) [inline Sources/Cosmos/Types/Simulation/Account.swift:2]: Let's fix or remove this header?

File: Sources/X/Auth/Ante/Ante.swift [hunk 2]
```diff
@@ -0,0 +1,8 @@
+//
+//  File.swift
```
Comment (paulofaria) [inline Sources/X/Auth/Ante/Ante.swift:2]: Same as previous comment!

File: Sources/X/Params/Subspace/ParameterSet.swift [hunk 3]
```diff
@@ -5,9 +5,10 @@ public typealias ValueValidatorFunction = (_ value: AnyEncodable) throws -> Void
 // ParamSetPair is used for associating paramsubspace key and field of param
 // structs.
 public struct ParameterSetPair {
-    let key: Data
-    let value: AnyEncodable
-    let validatorFunction: ValueValidatorFunction
+    #warning("TODO JAAP: does this need to be public?")
```
Comment (paulofaria) [inline Sources/X/Params/Subspace/ParameterSet.swift:8]: I think we should make everything that is not currently accessed elsewhere either private or internal and them expose them when necessary. Some properties will probably be appropriate as public, but I think we can ponder about this later. For now, let's expose the least possible.

**Review by paulofaria — APPROVED** [review]
Other than the file headers nit, looks good to me!

**Pull request merged**